proposed/http-message.md

```diff
@@ -1260,6 +1280,15 @@ interface UriInterface
      *
      * The string returned MUST omit the leading "#" character.
      *
+     * The value returned MUST be percent-encoded, but MUST NOT double-encode
+     * any characters. To determine what characters to encode, please refer to
+     * RFC 3986, Sections 2 and 3.5.
+     *
+     * As an example, if the fragment should include a hash symbol ("#"), that
+     * value MUST be encoded (e.g., "%23") when passed to the instance.
```

Comment:
So I guess for fragment we can just remove that sentence since there are no chars with special meaning in there.